### PR TITLE
Tls module revoke

### PR DESCRIPTION
**Pull request by quantonganh**
This is used to setup OpenVPN, for e.g:

```
openvpn_create_empty_crl_{{ instance }}:
  module:
    - wait
    - name: tls.create_empty_crl
    - ca_filename: 'ca'
    - ca_name: {{ ca_name }}
    - crl_file: '/etc/openvpn/{{ instance }}/crl.pem'
    - watch:
      - module: openvpn_ca
```

We cannot use an empty file as it is invaid CRL file which caused OpenVPN failed to start.

To revoke a certificate:

```
openvpn_revoke_client_cert_{{ r_client }}:
  module:
    - run
    - name: tls.revoke_cert
    - ca_name: {{ ca_name }}
    - ca_filename: 'ca'
    - CN: {{ r_client }}
    - cert_path: '/etc/openvpn/{{ instance }}/clients'
    - crl_file: '{{ config_dir }}/crl.pem'
    - require:
      - pkg: salt_minion_deps
      - module: openvpn_create_empty_crl_{{ instance }}
    - watch_in:
      - service: openvpn-{{ instance }}
```

### CODE REVIEW

File: salt/modules/tls.py [hunk 1]
```diff
@@ -1085,6 +1185,216 @@ def cert_info(cert_path, digest='sha256'):
...
+
+    crl_text = crl.export(ca_cert, ca_key)
+
+    if crl_file is None:
+        crl_file = '{0}/{1}/crl.pem'.format(
+                _cert_base_path(),
+                ca_name
+                )
+
+    if os.path.isdir('{0}'.format(crl_file)):
```
Comment (hvnsweeting) [inline salt/modules/tls.py]: is this different from just crl_file (without format)?

File: salt/modules/tls.py [hunk 2]
```diff
@@ -1085,6 +1185,216 @@ def cert_info(cert_path, digest='sha256'):
...
+                revoked = OpenSSL.crypto.Revoked()
+                revoked.set_serial(fields[3])
+                revoke_date_2_digit = datetime.strptime(fields[2], two_digit_year_fmt)
+                revoked.set_rev_date(revoke_date_2_digit.strftime(four_digit_year_fmt))
+                crl.add_revoked(revoked)
+
+    crl_text = crl.export(ca_cert, ca_key)
+
+    if crl_file is None:
+        crl_file = '{0}/{1}/crl.pem'.format(
```
Comment (hvnsweeting) [inline salt/modules/tls.py]: should this use ``os.path.join()`` instead?

Comment (hvnsweeting) [inline salt/modules/tls.py]: and also in many other places in this module

**Comment by jfindlay** [comment]
@quantonganh, there are some [lint errors](https://jenkins.saltstack.com/job/salt-pr-lint-n/5363/violations/file/salt/modules/tls.py/).

**Comment by quantonganh** [comment]
@jfindlay: it has been fixed.

**Comment by thatch45** [comment]
wow, yes, this module needs some serious os.path.join !

**Pull request merged**